utility/doc/PROTOCOL_CLIENT_SIDE_CHALLENGE.md

```diff
@@ -0,0 +1,213 @@
+  transactions.
+- Attacks on other node stakes in the situation that client side challenges
+  result in a net loss for the nodes who fail them.
+- Illegitimate reward farming for apps in the situation that client side
+  challenges result in a net gain for the app that executes the challenge.
+
+Additionally, to remain non-interactive, it needs to be the case that the
+relay request is inferable from the result of the relay response.
+
+### Client Side Challenge Mechanism
+
+At the time of whitelisting a relay chain for rewards, it will be required to
+able provide a pattern string that will pattern match to identify if a request
+is unable to be client-side challenged.
+
+This will require that the `pocketcore/SupportedBlockchains` parameter take
+the following form:
+
+```go
+type SupportedBlockchains struct {
+   Blockchains []SupportedBlockchain
+}
+
+type SupportedBlockchain struct {
+    RelayChainID string
+    NotChallengablePatterns []string
+}
+```
+
+The `NotChallengablePatterns` field is an array of pattern strings that should
+successfully match against the raw request string should if the request is not
+able to be client side challenged. This means having to make a change to what
+is expected of the app when requesting a Relay so that the App will send along
+a signed version of the string encoded HTTP request.
+
+```go
+type QueryRelayRequest stuct {
+    Payload       RelayPayload
+    SignedPayload string
+    Meta          RelayMetadata
+    Proof         RelayProof
+}
+```
+
+Then what is expected of the node when responding to a relay is the following
+`RelayResponse`, where the `Signature` field is now the `Signature` of the
+App's original signed payload appended with the `Response` string.
+
+```go
+type RelayResponse struct {
+	Signature string
+    Request   string
+	Response  string
+	Proof     RelayProof
+}
+```
+
+From here, the client side challenge is to be initiated from a transaction
+message sent with the following type:
+
+```go
+type ChallengeProofInvalidData struct {
+	MajorityResponses []RelayResponse
+	MinorityResponse  RelayResponse
+	ReporterAddress   Address
+}
+```
+
+From here, the following logical checks must be made:
+
+```mermaid
+stateDiagram-v2
+    ChallengeProofInvalidData --> MajorityCheck
+    MajorityCheck --> INVALID : No
+    MajorityCheck --> MajorityMatch : Yes
+    MajorityMatch --> INVALID : No
+    MajorityMatch --> MinorityDiff : Yes
+    MinorityDiff --> INVALID : No
+    MinorityDiff --> ActorsInSession : Yes
+    ActorsInSession --> INVALID : No
+    ActorsInSession --> IsRequestChallengable : Yes
+    IsRequestChallengable --> INVALID : No
+    IsRequestChallengable --> AppSignaturesMatch : Yes
+    AppSignaturesMatch --> INVALID : No
+    AppSignaturesMatch --> NodeSignaturesMatch : Yes
+    NodeSignaturesMatch --> INVALID : No
+    NodeSignaturesMatch --> SlashMinorityResponseNode : Yes
+```
+
+If it is the case that the challenge was a valid posted challenge, then the
```

Comment:
NIT: Curious to hear your thoughts. I realize this is personal writing style/preference, but this feels simpler to me. Wdyt?

```suggestion
If the posted challenge was valid, then the
```